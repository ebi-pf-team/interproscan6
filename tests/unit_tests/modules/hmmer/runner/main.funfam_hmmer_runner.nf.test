nextflow_process {

    name "Test Process FUNFAM_HMMER_RUNNER"
    script "interproscan/modules/hmmer/runner/main.nf"
    process "FUNFAM_HMMER_RUNNER"

    stage {
        copy "tests/unit_tests/test_inputs/hmmer_runners/hmmer_runner_fasta.faa"
        copy "tests/unit_tests/test_inputs/hmmer_runners/mock_hmm.hmm"
        copy "tests/unit_tests/test_inputs/hmmer_runners/cathSuperfamilies"
    }

    test("Should run without failures") {

        when {
            params {}
            process {
                """
                input[0] = [
                    file("meta/tests/unit_tests/test_inputs/hmmer_runners/hmmer_runner_fasta.faa"),
                    "unit-test",
                    file(".meta/tests/unit_tests/test_inputs/hmmer_runners/mock_hmm.hmm"),
                    ["switch"],
                    []
                ]
                input[1] = file("meta/tests/unit_tests/test_inputs/hmmer_runners/cathSuperfamilies")
                input[2] = "funfam"
                input[3] = true
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
