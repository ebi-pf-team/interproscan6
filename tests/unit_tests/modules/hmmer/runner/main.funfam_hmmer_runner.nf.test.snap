{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "funfam_unittest.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    [
                        
                    ]
                ],
                "2": [
                    "unit-test"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-20T18:00:08.578899"
    }
}